src/api/fields.py

```diff
@@ -16,9 +20,35 @@ def choice_strings_to_values(self):
         return {str(key): key for key in self.choices}
 
     def _get_choices(self):
-        return fields.flatten_choices_dict(self.grouped_choices)
+        if self._flattened_choices is None:
+            self._flattened_choices = fields.flatten_choices_dict(self.grouped_choices)
+        return self._flattened_choices
```

Comment:
I'm sure this is right, but just wondering if `_flattened_choices` is necessary, if everything is going back to `cached_choices`?

Reply:
the method gets called several times.